### ISSUE TITLE
DataGridColumn Filter issue

### ISSUE THREAD

**Issue by vmakharashvili** [stsrki/Blazorise#624]
**Describe the bug**
Every time I use this:
`<DataGridColumn  Filterable="true">`
I get errors in the browser console, which looks like this:

> blazor.webassembly.js:1 WASM: ﻿Unhandled exception rendering component:
>blazor.webassembly.js:1 WASM: Microsoft.JSInterop.JSException: element.addEventListener is not a function
>blazor.webassembly.js:1 WASM: TypeError: element.addEventListener is not a function

However, the filter function actually works.
I think it's a bug but maybe I don't use it properly.

Also, Maybe the DataGrid Columns needs to add server-side filtering.

**Comment by stsrki** [comment]
Not sure why it would give you an error. Filterable property is by default set to `true`. How do you handle your data? Do you read it from memory all at once, or do you load it dynamically on `ReadData` event?

**Comment by vmakharashvili** [comment]
I read it from memory all at once.
`<DataGrid TItem="Account"
                  Data="@_accounts"
                  @bind-SelectedRow="@_selectedAccount"
                  ShowPager="false"
                  Filterable="true"
                  RowDoubleClicked="(e => OnAccountClick(e.Item as Account))" Narrow="true">

            <DataGridColumn TItem="Account" Field="@nameof(Account.AccountNumber)" Caption="IBan" Sortable="true" Filterable="true">
            </DataGridColumn>
            <DataGridColumn TItem="Account" Field="@nameof(Account.Ccy)" Caption="Currency" Sortable="true" Filterable="true" /></DataGrid>`

This is how I do it.

**Comment by vmakharashvili** [comment]
In this case I have small data.

**Comment by stsrki** [comment]
Maybe one of your fields is a _complex_ type. For example is Ccy in `Account.Ccy` a string or is it a class or struct type? In case it is a complex type, then you need to define full "path". eg `Account.Ccy.Name`.

**Comment by vmakharashvili** [comment]
No. I commented out all the other columns and left just this AccountNumber column and still have the same errors in console.

**Comment by stsrki** [comment]
Can you please provide me with the small repro project so I can have a better look?

**Comment by vmakharashvili** [comment]
Thanks for your help. 

I played with my solution and found what was wrong. Now it work well.

I don't know is it a bug or not, but what I figured out is this:

my DataGrid was inside 'if' statement. If data wasn't null than DataGrid was generated. When I removed the if statement my browser console didn't show any error.

`@if(_data != null) {
  <DataGrid TItem="Account" Filterable="true" Data="@_data">
 <DataGridColumn TItem="Account" Field="@nameof(Account.AccountNumber)" 
                            Caption="IBan" Sortable="true" Filterable="true">
  ...
  </DataGrid>
}`
